### ISSUE TITLE
Failure when saving signatures with Rizzo

### ISSUE THREAD

**Issue by CreepNT** [fuzzywalls/ghidra_scripts#19]
When trying to save signatures from a PS2 ELF, located in a shared project, the script appears to run for a while then crashes with the following error :
```py
RizzoSave.py> Running...
Building Rizzo signatures, this may take a few minutes...
Traceback (most recent call last):
  File "C:\Users\CreepNT\ghidra_scripts\RizzoSave.py", line 15, in <module>
    rizz = rizzo.Rizzo(currentProgram)
  File "__pyclasspath__/utils/rizzo.py", line 290, in __init__
  File "__pyclasspath__/utils/rizzo.py", line 612, in _generate
  File "__pyclasspath__/utils/rizzo.py", line 589, in _hash_function
  File "__pyclasspath__/utils/rizzo.py", line 556, in _hash_block
AttributeError: 'NoneType' object has no attribute 'isData'
RizzoSave.py> Finished!
```
As I would expect, no signature files are exported after this crash.

Ghidra Version : 9.2.2
Scripts located in `C:\Users\CreepNT\ghidra_scripts`

Any idea what could be cause this failure ? Maybe I'm missing a library ?

**Comment by fuzzywalls** [comment]
You shouldn’t be missing any libraries as all the scripts are self contained. Any chance you could attach the ELF in question or give me a link to it? I’ll look into it tonight (or more likely) early next week

**Comment by CreepNT** [comment]
Sure enough, thanks a lot for having a look.
[SCUS_974.65](https://github.com/tacnetsol/ghidra_scripts/files/6407123/SCUS_974.65.zip)

**Comment by CreepNT** [comment]
A thought just crossed my head :
since this is a PS2 ELF, I had to use a plugin that adds a format (namely [Ghidra-EmotionEngine](https://github.com/beardypig/ghidra-emotionengine)).

Could the issue be related to this ?

**Comment by fuzzywalls** [comment]
It could. I’ll include it in my testing, thanks.

**Comment by fuzzywalls** [comment]
@CreepNT I did some digging into the problem. I think it stems from a bad function in the Ghidra-EmotionEngine project, and put a ticket in for them to fix it (if it is actually a problem) [here](https://github.com/beardypig/ghidra-emotionengine/issues/58). In the mean time I made a fix to Rizzo that will allow you to use it as is. It shouldn't be a problem unless they fix the ticket I submitted, at that point you will probably need to regenerate your Rizzo signatures.
